File: labos_03/zadatak 1.1.1-1.1.4/parrot.c
#include <stdio.h>
#include <stdlib.h>


typedef char const* (*PTRFUN)();

struct Parrot{
    PTRFUN *vtable;
    char const* name;
};

char const* name(void* this){
    return ((struct Parrot*)this)->name;
}


char const* parrotGreet(void){
  return "Sto mu gromova!";
}

char const* parrotMenu(void){
  return "brazilske orahe";
}

PTRFUN parrotVTable[3]= {name,parrotGreet,parrotMenu};

void constructParrot(struct Parrot* parrot,char* name){
    parrot->name=name;
    parrot->vtable=parrotVTable;
}

void* create(char* name){
  struct Parrot* parrot =  malloc(sizeof(struct Parrot));
  constructParrot(parrot,name);
  return parrot;
}

File: labos_03/zadatak_02/src/main/java/hr/fer/zemris/ooup/lab2/model/plugins/Parrot.java
package hr.fer.zemris.ooup.lab2.model.plugins;

import hr.fer.zemris.ooup.lab2.model.Animal;

public class Parrot extends Animal{

    private final String name;

    public Parrot(String name){
        this.name=name;
    }

    @Override
    public String name() {
        return name;
    }

    @Override
    public String greet() {
        return "Sto mu gromova!";
    }

    @Override
    public String menu() {
        return "brazilske orahe";
    }

}
